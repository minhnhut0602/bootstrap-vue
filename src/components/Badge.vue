<template>
    <div
        class="badge"
        :class="classes"
    >
        <slot />
    </div>
</template>

<script lang="ts" setup>
import {computed, defineProps} from 'vue';
import useVariant, {variantProps} from '@/composables/useVariant';
import useClasses from '@/composables/useClasses';

const props = defineProps({
    pill: {
        type: Boolean,
        default: false,
    },
    ...variantProps,
});

const {classes} = useClasses([
    computed(() => props.pill ? 'rounded-pill' : undefined).value,
    useVariant(props.variant, 'bg-{0}').variantClass.value,
]);
</script>
