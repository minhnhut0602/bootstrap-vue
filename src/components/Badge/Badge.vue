<template>
    <div
        class="badge"
        :class="classes"
    >
        <slot />
    </div>
</template>

<script lang="ts">
import useVariant, {variantProps} from '../../composables/useVariant';
import {computed} from 'vue';
import useClasses from '../../composables/useClasses';
</script>

<script lang="ts" setup>
const props = defineProps({
    pill: {
        type: Boolean,
        default: false,
    },
    ...variantProps,
});

const {classes} = useClasses(computed(() => [
    computed(() => props.pill ? 'rounded-pill' : undefined).value,
    useVariant(props.variant, 'bg-{0}').variantClass.value,
]));
</script>
