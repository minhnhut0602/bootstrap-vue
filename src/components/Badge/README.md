# Badge

```html
<Badge
    pill
    variant="danger"
>
    I am a badge
</Badge>
```

## Props

| Name | Type | Description |
| - | - | - |
| pill | boolean | Makes the badge more rounded. |
| variant | [Variant](#link-to-variant-definition) (string) | The variant of the badge. |

## Slots

| Name | Description |
| - | - |
| default | Displays content inside the badge. |
