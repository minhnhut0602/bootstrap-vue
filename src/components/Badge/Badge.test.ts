import Badge from '.';
import {shallowMount} from '@vue/test-utils';

describe('template', () => {
    componentRenderTest(Badge);

    componentSlotRenderTest(Badge);

    componentWrapperClassTest(Badge, {pill: true}, 'rounded-pill');

    componentWrapperClassTest(Badge, {variant: 'danger'}, 'bg-danger');
});
