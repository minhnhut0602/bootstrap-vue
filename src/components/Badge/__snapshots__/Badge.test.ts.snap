// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template renders default 1`] = `"<div class=\\"badge bg-primary\\"></div>"`;
